librz/arch/il/analysis_il.c

```diff
@@ -280,6 +299,53 @@ RZ_API RzAnalysisILStepResult rz_analysis_il_vm_step_while(RZ_NONNULL RzAnalysis
 	return res;
 }
 
+/**
+ * \brief Repeatedly perform steps in the VM until the \p cond callback returns false
+ *
+ * \param analysis Pointer to an RzAnalysis struct, likely representing the analysis context.
+ * \param vm Pointer to an RzAnalysisILVM struct, representing the IL virtual machine to be stepped.
+ * \param reg Optional pointer to an RzReg struct, potentially holding register values to be used during the step.
+ * \param cond Pointer to a function that determines the loop's continuation condition. This function takes two arguments:
+ *          * vm: Pointer to the same RzAnalysisILVM struct passed to rz_analysis_il_vm_step_while.
+ *          * user: Pointer to user-provided data that can be used by the condition function.
+ * \param user Pointer to user-defined data that can be passed to the condition function.
+ *
+ * \return RZ_ANALYSIS_IL_STEP_RESULT: Enumeration value indicating the outcome of the stepping operation.
+ *         Possible values (implementation specific):
+ *             - RZ_ANALYSIS_IL_STEP_OK: Successful execution of the while loop step.
+ *             - RZ_ANALYSIS_IL_STEP_ERROR: Encountered an error during execution.
+ *             - RZ_ANALYSIS_IL_STEP_INVALID: Invalid arguments or state resulted in undefined behavior.
+ */
+RZ_API RzAnalysisILStepResult rz_analysis_il_vm_step_while(
+	RZ_NONNULL RzAnalysis *analysis, RZ_NONNULL RzAnalysisILVM *vm, RZ_NULLABLE RzReg *reg,
+	RZ_NONNULL cond_callback cond, RZ_NULLABLE void *user) {
```

Comment:
If `cond_callback` is used in the `RZ_API` then is worth moving it with a proper name into the header.